src/java/org/apache/cassandra/journal/Flusher.java

```diff
@@ -305,7 +305,7 @@ private void asyncFlushPeriodic(ActiveSegment<K>.Allocation ignore, Executor exe
                 pending.decrementAndGet();
             }
             written.incrementAndGet();
-            executor.execute(onDurable);
+            executor.execute(callback);
```

Comment:
rather than `AsyncWriteCallback extends Runnable` it might be better to do `callback::onSuccess`

Reply:
^ see above

Reply:
Resolving as above ^